src/AclApi.ts

```diff
@@ -92,17 +101,45 @@ class AclApi {
     const response = await this.fetch(this.aclUrl, {
       method: 'PUT',
       headers: {
-        'Content-Type': 'text/turtle'
+        [CONTENT_TYPE]: 'text/turtle',
+        ...(this.eTag && { [IF_MATCH]: this.eTag }) // Check that the acl file hasn't been modified
       },
       body: turtle
     })
+
     if (!response.ok) {
       console.error(response)
+      if (response.status === 416) {
+        throw new Error('Error while trying to save the acl file: The file has been changed by another program')
+      }
       throw new Error(`Error while trying to save the acl file: ${response.status} - ${response.statusText}`)
     }
+
+    this.eTag = response.headers.get(ETAG)
```

Comment:
It's cleaner to store the Etag without the surrounding `"`. So if the header is:

`ETag: "bla"`

Then you want to store `'bla'` and not `'"bla"'`.